src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -97,6 +97,44 @@ public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage>? predica
         return this;
     }
 
+    /// <summary>
+    /// Configures the builder to match any request that meets the criteria defined by the specified predicate.
+    /// </summary>
+    /// <param name="predicates">
+    /// Two or more delegates to a method which returns <see langword="true"/> if the
+    /// request is considered a match; otherwise <see langword="false"/>.
+    /// </param>
+    /// <returns>
+    /// The current <see cref="HttpRequestInterceptionBuilder"/>.
+    /// </returns>
+    /// <remarks>
+    /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
+    /// </remarks>
+    public HttpRequestInterceptionBuilder For(params Predicate<HttpRequestMessage>[] predicates)
```

Comment:
Change this to a `Func<HttpRequestMessage, Task<bool>` so that it supports asynchronous cases, such as reading the request body (which needs async method calls) to determine a match.

Alternatively add a second overload for that type of predicate and leave this one as-is.